### DISCUSSION TITLE
Is there a way to identify the packet type?

### DISCUSSION THREAD

**Discussion by tbazen** [chmorgan/sharppcap#330] [Q&A] [answered]
In my scenario is I have `pcap`  file generated from wireshark and I would like to parse each packet depending on the type of the packet. more generally the packets can be parsed as follows ...
```
var packet = Packet.ParsePacket(LinkLayers.Ethernet, packetByte);
var ethernetPacket = parsedPacket.Extract<EthernetPacket>();
```
In this case I know before hand the packet is of type `EthernetPacket`, hence I can use the `Extract` method confidently. is there a way of telling the packet type so that I can call the `Extract` method more dynamically/conditionally depending packet type?

Thanks in advance

**Comment by kayoub5** [comment, marked as answer]
LinkLayer will tell what the first layer is, after that each protocol will have its own way for telling what the next protocol is, for example in ethernet the field ethertype tells you what the next protocol is.
